File: VideoSplashScreen/app/src/main/java/com/blackbirds/shakil/videosplashscreen/MainActivity.java
package com.blackbirds.shakil.videosplashscreen;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.AppCompatButton;

import android.content.Intent;
import android.graphics.Color;
import android.media.MediaPlayer;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;
import android.view.WindowManager;
import android.widget.ImageView;
import android.widget.VideoView;

public class MainActivity extends AppCompatActivity {

    VideoView videoView;
    AppCompatButton btnNextActivity;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        requestWindowFeature(1);
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS, WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS);
        getWindow().setStatusBarColor(Color.TRANSPARENT);
        setContentView(R.layout.activity_main);

        videoView = findViewById(R.id.videoView);
        btnNextActivity = findViewById(R.id.btnNextActivity);

        String path = "android.resource://" + getPackageName() + "/" + R.raw.video;
        Uri uri = Uri.parse(path);
        videoView.setVideoURI(uri);
        videoView.start();

        /*videoView.setOnPreparedListener(mp -> {
            mp.start();
            //imageView.setVisibility(View.GONE);
        });*/

        /** If you want video was play again and again */
        videoView.setOnCompletionListener(mp -> {

            if (isFinishing()){
                return;
            }
            startActivity(new Intent(this, SecondActivity.class));
            finish();
        });

        btnNextActivity.setOnClickListener(v -> {
            startActivity(new Intent(this, SecondActivity.class));
        });
    }
}

File: KotlinVideoSplashScreen/app/src/main/java/com/blackbirds/shakil/kotlinvideosplashscreen/MainActivity.kt
package com.blackbirds.shakil.kotlinvideosplashscreen

import android.content.Intent
import android.graphics.Color
import android.net.Uri
import android.os.Bundle
import android.view.WindowManager
import android.widget.VideoView
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.AppCompatButton

class MainActivity : AppCompatActivity() {

    var videoView: VideoView? = null
    var btnNextActivity: AppCompatButton? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        requestWindowFeature(1)
        window.setFlags(
            WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS,
            WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS
        )
        window.statusBarColor = Color.TRANSPARENT
        setContentView(R.layout.activity_main)

        videoView = findViewById(R.id.videoView)
        btnNextActivity = findViewById(R.id.btnNextActivity)

        val path = "android.resource://" + packageName + "/" + R.raw.video
        val uri = Uri.parse(path)
        videoView!!.setVideoURI(uri)
        videoView!!.start()

        videoView!!.setOnCompletionListener {
            if (isFinishing) {
                true
            }
            startActivity(Intent(this, SecondActivity::class.java))
            finish()
        }
    }
}